lib/runner/VirtualDeviceInvoker.js

```diff
@@ -90,6 +90,17 @@ module.exports = class VirtualDeviceInvoker extends Invoker {
         }
     }
 
+    beforeTest(test) {
```

Comment:
Adding the logic in this event will cause an unnecessary wait for all utterances, if one of those would fail.
Instead since this only affects the cases where asyncMode: true , the logic should be added if the utterance is PAUSE in the batchAsyncInvocation method